firedrills.md

```diff
@@ -0,0 +1,782 @@
+
+```bash
+node-1# kubectl cordon 192.168.121.217
+node/192.168.121.217 uncordoned
+```
+
+### Scenario 2.2: Replacing Failed Node
+
+Now let’s consider another scenario - a failed node. A node can fail for various reasons, for the sake of this exercise we will assume that the node is unrecoverable. Let’s simulate this scenario by wiping out the node clean:
+
+```bash
+node-3# sudo gravity system uninstall --confirm
+```
+
+!!! warning:
+    This command is equivalent of `rm -rf` for all Kubernetes/Gravity data so never run it on a production cluster.
+
+As we found out before, when we were talking about etcd, a 3-node cluster can afford to lose 1 node so the cluster remains functional, but the node is shown as offline:
+
+```bash
+node-1# gravity status
+node-1# kubectl get nodes
+node-1# sudo gravity exec etcdctl cluster-health
+```
+
+To replace the failed node, we first need to remove it from the cluster. To do that, let’s run:
+
+```bash
+node-1# sudo gravity remove node-3 --force
+Please confirm removing node-3 (192.168.121.217) from the cluster (yes/no):
+```
+
+Note that we need to force-remove in this case because the node is offline. For online nodes the flag is not required.
+
+If you launch `gravity status` command now, you will see that Gravity launched “shrink” operation that is working on properly evicting the node from the cluster and Kubernetes state.
+
+```bash
+node-1# gravity status
+...
+Active operations:
+   * operation_shrink (71266b23-d0ea-46b6-a5c4-15906eeb770a)
+     started:  Tue Apr 30 21:34 UTC (12 seconds ago)
+     unregistering the node, 10% complete
+```
+
+Once the operation completes, the cluster will switch back into the active state and the node will no longer appear in the gravity status. The `kubectl get nodes` and `etcdctl cluster-health` commands will also show that the node is no longer a part of the cluster.
+
+```bash
+node-1# gravity status
+...
+Cluster nodes:  nostalgicbell4909
+   Masters:
+       * node-1 (192.168.121.232, node)
+           Status:     healthy
+       * node-2 (192.168.121.198, node)
+           Status:     healthy
+```
+
+Now that the node has been properly removed and the cluster is healthy (but running at a reduced capacity of 2 nodes), we can join a replacement node. In our case, we can just join back the node we’ve just removed.
+
+First, let’s find out the cluster’s join token which the joining node will need to provide to authenticate:
+
+```bash
+node-1# gravity status --token
+45b56f472e71
+```
+
+Next, we need to run a `gravity join` command on our new node, however when we ran system uninstall on it, the gravity binary was removed as well. Let’s get it back:
+
+```bash
+node-3# curl -k -H "Authorization: Bearer 45b56f472e71" https://192.168.121.232:3009/portal/v1/gravity -o gravity && chmod +x gravity
```

Comment:
Did this change? I think downloading the binary is an unauthenticated endpoint and can be done without providing the token.